tests/foreman/api/test_contentview.py

```diff
@@ -571,7 +571,7 @@ def test_negative_publish_during_repo_sync(self, content_view, module_target_sat
         task_search = module_target_sat.api.ForemanTask().search(
             query={'search': f'{task_action} and started_at >= "{timestamp}"'}
         )
-        assert len(task_search) == 1
+        assert len(task_search) > 0
```

Comment:
You can get the task id of  publish task and search for it. It will be unique and you will get a single task.

Reply:
@shweta83 Also tried asynchronous Publish, it also returns `None` once exception is raised/caught.
Instead I've updated to narrow the search scope, for the publish task with the desired `cv.name` and `org.name`. 
Let me know if this addresses your suggestion!

Reply:
in lines below, I get id from first entry in `task_search`, which returns like the output in your comment.
Then I poll that task by id and get failed output.
Am I missing something else or some other step for getting task id, other than `ForemanTask.search()`?